components/public-api/gitpod/v1/workspace.proto

```diff
@@ -74,71 +83,43 @@ message ListWorkspacesRequest {
 }
 
 message ListWorkspacesResponse {
-  // workspaces are the workspaces that matched the query
-  repeated Workspace workspaces = 1;
-
   // pagination contains the pagination options for listing workspaces
-  PaginationResponse pagination = 2;
+  PaginationResponse pagination = 1;
+
+  // workspaces are the workspaces that matched the query
+  repeated Workspace workspaces = 2;
 }
 
+// Required fields:
+// - metadata.organization_id
+// - metadata.configuration_id
 message CreateAndStartWorkspaceRequest {
-  message Git {
-    // clone_url is the URL of the repository to clone
-    string clone_url = 1;
-
-    // ref is an alternatively symbolic. e.g. refs/tags/v1.0,
-    // empty string means the default branch of the repository
-    string ref = 2;
+  message ContextURL {
+    // url is the URL from which the workspace is created
+    string url = 1;
 
-    // create_local_branch is the branch you want to create based on provided
-    // clone_url and ref when workspace started
-    string create_local_branch = 3;
+    // workspace_class is the class of the workspace we ought to start
+    string workspace_class = 2;
   }
 
-  // organization_id is the ID of the organization to create the workspace
-  //
-  // +required
-  string organization_id = 1;
+  // metadata is data associated with this workspace that's required for other parts of Gitpod to function
+  WorkspaceMetadata metadata = 2;
 
-  // configuration_id is the ID of the configuration to use
-  string configuration_id = 2;
-
-  // source describes the source refer of workspace.
-  //
-  // +required
   oneof source {
-    // git describes the source refer of workspace
-    // Obtain available git using the ContextService.ParseContext operation if
-    // not sure about it.
-    Git git = 3;
-
-    // context_url is for backward compatiblity with the current dashboard, use
-    // ContextService.ParseContext get get a Git source instead
-    string context_url = 4 [deprecated = true];
-  }
-
-  // additional_env_variables provide additional environment variables to the
-  // workspace.
-  // It will take precedence over environment variables provided by
-  // the user and the configuration
-  repeated WorkspaceEnvironmentVariable additional_env_variables = 5;
-
-  string region = 6;
+    // context_url is the URL from which the workspace is created
+    ContextURL context_url = 3;
 
-  // workspace_class is the class of the workspace
-
-  string workspace_class = 7;
-
-  EditorReference editor = 8;
-
-  string name = 9;
+    // spec is the configuration of the workspace that's required for the to start the workspace
+    WorkspaceSpec spec = 4;
+  }
 
-  bool pinned = 10;
+  // editor specifies the editor that will be used with this workspace.
+  EditorReference editor = 5;
```

Comment:
@mustard-mh please move it in ContextURL , because in case of WorkspaceSpec there is already EditorReference